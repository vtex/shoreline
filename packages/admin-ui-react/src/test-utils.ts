import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
