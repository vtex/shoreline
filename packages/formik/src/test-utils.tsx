import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
