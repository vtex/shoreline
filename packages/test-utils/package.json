{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
