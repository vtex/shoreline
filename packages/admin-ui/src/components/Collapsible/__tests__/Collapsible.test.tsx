import React from 'react'
import { render, withState } from '../../../test-utils'

import {
  Collapsible,
  useCollapsibleState,
  CollapsibleHeader,
  CollapsibleContent,
} from '../index'

const StatefulCollapsible = withState(Collapsible, () =>
  useCollapsibleState({ visible: true })
)

describe('Collapsible', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <StatefulCollapsible data-testid="collapsible" csx={{ bg: 'azure' }}>
        <CollapsibleHeader
          label="Collapsible Label"
          data-testid="header"
          csx={{ bg: 'azure' }}
        />
        <CollapsibleContent data-testid="content" csx={{ bg: 'azure' }}>
          Collapsible Content
        </CollapsibleContent>
      </StatefulCollapsible>
    )

    expect(getByTestId('collapsible')).toHaveStyleRule('background', 'azure')
    expect(getByTestId('header')).toHaveStyleRule('background', 'azure')
    expect(getByTestId('content')).toHaveStyleRule('background', 'azure')
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <StatefulCollapsible>
        <CollapsibleHeader label="Collapsible Label" />
        <CollapsibleContent>Collapsible Content</CollapsibleContent>
      </StatefulCollapsible>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with nesting collapsibles', () => {
    const { asFragment } = render(
      <StatefulCollapsible>
        <CollapsibleHeader label="Collapsible Label" />
        <CollapsibleContent>
          <StatefulCollapsible>
            <CollapsibleHeader label="Collapsible Label" />
            <CollapsibleContent>Collapsible Content</CollapsibleContent>
          </StatefulCollapsible>
        </CollapsibleContent>
      </StatefulCollapsible>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
