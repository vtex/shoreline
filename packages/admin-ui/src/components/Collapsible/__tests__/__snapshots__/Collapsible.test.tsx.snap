// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible should match snapshot 1`] = `
<DocumentFragment>
  .emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
  color: #000000;
  background: transparent;
}

.emotion-3:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-3:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-3:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-3 svg {
  height: 20px;
  width: 20px;
  min-height: 20px;
  max-height: 20px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
  -webkit-transition: -webkit-transform 150ms ease;
  transition: transform 150ms ease;
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}



.emotion-8 {
  background: #FFFFFF;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 4px;
  border-width: 1px;
}

.emotion-5 {
  padding: 1.5rem;
  padding-left: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-5 div>button:nth-of-type(n+2) {
  margin-left: 0.25rem;
}

.emotion-6 {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-bottom: 1rem;
}

.emotion-6 .__admin-ui-collapsible--header {
  padding: 1rem;
  padding-left: 0rem;
}

.emotion-6 .__admin-ui-collapsible--content {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-bottom: 1rem;
}

<div
    class="emotion-8"
  >
    <header
      class="emotion-5 __admin-ui-collapsible--header"
    >
      <button
        aria-controls="react-collapsed-panel-3"
        aria-expanded="true"
        class="emotion-3"
        id="react-collapsed-toggle-3"
        role="button"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-1"
          >
            <svg
              class="emotion-0"
              fill="currentColor"
              height="1.25rem"
              viewBox="0 0 256 256"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16rem"
                width="16rem"
              />
              <polyline
                fill="none"
                points="48 160 128 80 208 160"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
              />
            </svg>
             Collapsible Label
          </div>
        </div>
      </button>
      <div
        class="emotion-4"
      />
    </header>
    <div
      aria-hidden="false"
      class="emotion-7"
      id="react-collapsed-panel-3"
      style="box-sizing: border-box;"
    >
      <div
        class="__admin-ui-collapsible--content emotion-6"
      >
        Collapsible Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Collapsible should match snapshot with nesting collapsibles 1`] = `
<DocumentFragment>
  .emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: auto;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
  color: #000000;
  background: transparent;
}

.emotion-3:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-3:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-3:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

.emotion-3 svg {
  height: 20px;
  width: 20px;
  min-height: 20px;
  max-height: 20px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
  -webkit-transition: -webkit-transform 150ms ease;
  transition: transform 150ms ease;
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}



.emotion-14 {
  background: #FFFFFF;
  border: 1px solid hsla(0, 0%, 87%, 1);
  border-radius: 4px;
  border-width: 1px;
}

.emotion-5 {
  padding: 1.5rem;
  padding-left: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-5 div>button:nth-of-type(n+2) {
  margin-left: 0.25rem;
}

.emotion-12 {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-bottom: 1rem;
}

.emotion-12 .__admin-ui-collapsible--header {
  padding: 1rem;
  padding-left: 0rem;
}

.emotion-12 .__admin-ui-collapsible--content {
  padding-left: 1rem;
  padding-right: 1rem;
  padding-bottom: 1rem;
}

<div
    class="emotion-14"
  >
    <header
      class="emotion-5 __admin-ui-collapsible--header"
    >
      <button
        aria-controls="react-collapsed-panel-4"
        aria-expanded="true"
        class="emotion-3"
        id="react-collapsed-toggle-4"
        role="button"
        tabindex="0"
        type="button"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-1"
          >
            <svg
              class="emotion-0"
              fill="currentColor"
              height="1.25rem"
              viewBox="0 0 256 256"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16rem"
                width="16rem"
              />
              <polyline
                fill="none"
                points="48 160 128 80 208 160"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="18"
              />
            </svg>
             Collapsible Label
          </div>
        </div>
      </button>
      <div
        class="emotion-4"
      />
    </header>
    <div
      aria-hidden="false"
      class="emotion-13"
      id="react-collapsed-panel-4"
      style="box-sizing: border-box;"
    >
      <div
        class="__admin-ui-collapsible--content emotion-12"
      >
        <div
          class="emotion-14"
        >
          <header
            class="emotion-5 __admin-ui-collapsible--header"
          >
            <button
              aria-controls="react-collapsed-panel-5"
              aria-expanded="true"
              class="emotion-3"
              id="react-collapsed-toggle-5"
              role="button"
              tabindex="0"
              type="button"
            >
              <div
                class="emotion-2"
              >
                <div
                  class="emotion-1"
                >
                  <svg
                    class="emotion-0"
                    fill="currentColor"
                    height="1.25rem"
                    viewBox="0 0 256 256"
                    width="1.25rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="16rem"
                      width="16rem"
                    />
                    <polyline
                      fill="none"
                      points="48 160 128 80 208 160"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="18"
                    />
                  </svg>
                   Collapsible Label
                </div>
              </div>
            </button>
            <div
              class="emotion-4"
            />
          </header>
          <div
            aria-hidden="false"
            class="emotion-13"
            id="react-collapsed-panel-5"
            style="box-sizing: border-box;"
          >
            <div
              class="__admin-ui-collapsible--content emotion-12"
            >
              Collapsible Content
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
